lib/tcpip_stack_direct.ml

```diff
@@ -77,10 +77,15 @@ struct
   let ipv4 { ipv4; _ } = ipv4
 
   let listen_udpv4 t ~port callback =
-    Hashtbl.replace t.udpv4_listeners port callback
+    if port < 0 || port > 65535
+    then raise (Failure "invalid port number")
```

Comment:
The error message should be factored out into something like this
```ocaml
let err_invalid_port p = Printf.sprintf "invalid port number (%d)" p
```
And the raised exception should be `Invalid_argument`.